smart-contracts/contracts/hooks/GitcoinHook.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@unlock-protocol/contracts/dist/PublicLock/IPublicLockV13.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract GitcoinHook is Ownable {
+  mapping(address => bool) public signers;
+
+  constructor() {}
+
+  function addSigner(address signer) public onlyOwner {
+    signers[signer] = true;
+  }
+
+  function removeSigner(address signer) public onlyOwner {
+    signers[signer] = false;
+  }
+
+  /**
+   * Price is the same for everyone... but we fail if signature by Unlock Lab's backend service (sent as signature) does not match!
+   * The Unlock Labs backend service verifies that the purchaser has a valid Gitcoin passport and signs the address of the recipient to confirm.
+   */
+  function keyPurchasePrice(
+    address /* from */,
+    address recipient,
+    address /* referrer */,
+    bytes calldata signature /* data */
+  ) external view returns (uint256 minKeyPrice) {
+    string memory message = toString(recipient);
+    require(checkIsSigner(message, signature), "WRONG_SIGNATURE");
+    if (address(msg.sender).code.length > 0) {
+      return IPublicLockV13(msg.sender).keyPrice();
+    }
+    return 0;
```

Comment:
We should probably never return `0` here because that means the user could pay `0` instead.
I think a safer approach is just to revert here?

Reply:
you're right, that was an oversight. great observation! now fixed!